### DISCUSSION TITLE
Replacing CC2531 for CC2652RB > sluggish/crashing

### DISCUSSION THREAD

**Discussion by JoshuaLoman** [Koenkk/zigbee2mqtt#10460] [General]
Hi all, pulling my hairs on this!

I am running a pi 3b+ with a CC2531 for a while now. It was running pretty smooth actually, the overal response of my zigbee devices was pretty fast and reliable. I even replaced the CC2531 stick once successfully without having to re-pair my devices.

Because of hitting the limit of devices, i wanted to upgrade to a better adapter, so i bought the slaesh CC2652RB. Yesterday i have tried to replace the adapter without having to re-pair all my devices.

I have tried to write the old IEEE address to the CC2652RB adapter with the ZigStar software on windows, but for some reason it did not work. I tried to put the stick in BSL mode by holding down the BL button, put the stick in and press the RST button. After that i could read the current IEEE address, but rewriting it with the old one didnt work.
As mentioned, al also tried to reflash the stick with the ZigStar software and rewriting the IEEE address at the same time. It looked like that the flashing worked, because the progress bar showed the progress of flashing.
I used the latest coordinator software: CC2652RB_coordinator_20210708.hex

After that, the IEEE address was still the old address. Because replacing the CC2531 without repairing worked before, i thought i would try to replace the stick again now without rewriting the IEEE address.

I replaced the stick, at first some errors, but after a few HA restarts and Zigbee2MQTT reboots, i looked like it was starting to work. All devices started working, without repairing.
But at some moments the response time was pretty fast, but that turned pretty quick. For example, i could turn a light off with a quick reponse, but after that it just looked like Z2M hangs for 10-30 seconds before i could switch anything. This did not go better after a full host reboot or unplugging and plugging in the adapter again.
After about an hour, i again tried to restart Z2M, and started to throw me errors and stopped working.

I put my CC2531 stick back in and restored a backup, after that everything is working again, with good responses like before.

Anyone has an idea?

**Comment by lux73** [comment]
i would recommend to reflash your cc2652 & re-pair all Devices - i have done this after switching from cc2531 to my new cc2652

never had any glitch after :-)

**Reply by JoshuaLoman** [reply]
That would be too bad.. I bought the cc2652rb just to not repair my devices 😆
